packages/server/modules/notifications/services/queue.ts

```diff
@@ -0,0 +1,173 @@
+import { UninitializedResourceAccessError } from '@/modules/shared/errors'
+import { Optional } from '@/modules/shared/helpers/typeHelper'
+import {
+  InvalidNotificationError,
+  NotificationValidationError,
+  UnhandledNotificationError
+} from '@/modules/notifications/errors'
+import {
+  isNotificationMessage,
+  NotificationHandler,
+  NotificationMessage,
+  NotificationType,
+  NotificationTypeHandlers
+} from '@/modules/notifications/helpers/types'
+import { notificationsDebug } from '@/modules/shared/utils/logger'
+import { isProdEnv, isTestEnv } from '@/modules/shared/helpers/envHelper'
+import Bull from 'bull'
+import { buildBaseQueueOptions } from '@/modules/shared/helpers/bullHelper'
+
+export type NotificationJobResult = {
+  status: NotificationJobResultsStatus
+  type: NotificationType | undefined
+}
+
+export enum NotificationJobResultsStatus {
+  Success = 'success',
+  ValidationError = 'validation-error'
+}
+
+const handlers = new Map<NotificationType, NotificationHandler>()
+
+export const NOTIFICATIONS_QUEUE_MAIN = `default:user-notifications`
+export const NOTIFICATIONS_QUEUE_TEST = `test:user-notifications`
+
+export const NOTIFICATIONS_QUEUE = isTestEnv()
+  ? NOTIFICATIONS_QUEUE_TEST
+  : NOTIFICATIONS_QUEUE_MAIN
+
+let queue: Optional<Bull.Queue>
+const buildNotificationsQueue = (queueName: string) =>
+  new Bull(queueName, {
+    ...buildBaseQueueOptions(),
+    ...(!isTestEnv()
+      ? {
+          limiter: {
+            max: 10,
+            duration: 1000
+          }
+        }
+      : {}),
+    defaultJobOptions: {
+      attempts: 1,
+      timeout: 10 * 1000, // 10s execution timeout
+      removeOnComplete: isProdEnv(),
```

Comment:
@gjedlicska all finished jobs are removed from the queue in prod so as to not fill it up, but that means we can't check them out afterwards as some kind of log output. if we need more insight into what's going on in the queues, we can re-think this